gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMessageHttpRequestFormatter.java

```diff
@@ -56,20 +65,19 @@
   }
 
   @Override
-  public Map<String, String> getPathParams(ApiMessage apiMessage, Set<String> paramNames) {
-    Map<String, String> pathParams = new HashMap<>();
-    Map<String, List<String>> pathParamMap = apiMessage.populateFieldsInMap(paramNames);
-    Iterator<Map.Entry<String, List<String>>> iterator = pathParamMap.entrySet().iterator();
-    while (iterator.hasNext()) {
-      Map.Entry<String, List<String>> pair = iterator.next();
-      pathParams.put(pair.getKey(), pair.getValue().get(0));
+  public Map<String, String> getPathParams(T apiMessage, String resourceNameField)
+      throws IOException {
+    String resourceNamePath = apiMessage.getFieldStringValue(resourceNameField);
+    if (resourceNamePath == null) {
+      throw new IOException(
+          String.format("Resource name field %s is null in message object.", resourceNameField));
```

Comment:
This isn't an `IOException` - it should be `IllegalArgumentException`.

Reply:
Done.